agent/consul/state/config_entry.go

```diff
@@ -281,7 +292,42 @@ func (s *Store) EnsureConfigEntryCAS(idx, cidx uint64, conf structs.ConfigEntry)
 		return false, nil
 	}
 
-	if err := ensureConfigEntryTxn(tx, idx, conf); err != nil {
+	if err := ensureConfigEntryTxn(tx, idx, false, conf); err != nil {
+		return false, err
+	}
+
+	err = tx.Commit()
+	return err == nil, err
+}
+
+// EnsureConfigEntryStatusCAS is called to do a check-and-set upsert of a given config entry status.
+func (s *Store) EnsureConfigEntryStatusCAS(idx, cidx uint64, conf structs.ConfigEntry) (bool, error) {
```

Comment:
The name and godoc here imply to me that we're _only_ setting the status; however, I believe we're setting the whole config entry _and the status_ where we would usually just ignore any status on a normal `EnsureConfigEntryCAS`. Is that accurate?

If so, I might change the name a bit and update the godoc to reflect
```suggestion
// EnsureConfigEntryWithStatusCAS is called to do a check-and-set upsert of a given config entry and its status.
func (s *Store) EnsureConfigEntryWithStatusCAS(idx, cidx uint64, conf structs.ConfigEntry) (bool, error) {
```

Reply:
It might also be helpful to note in the godocs for other `EnsureX` that the status is ignored

Reply:
That makes sense RE updating the other godoc code. The name reflects my own indecision about whether or not this method should be used as a generic internal update method mechanism that can update both spec and status all at once or if it should solely be scoped to the status. I had followed the naming similar to the kubernetes client with `UpdateStatus`, but then went ahead and just implemented it as an update with the `Status` resetting bypassed.

Though Kubernetes does scope things solely to updating the status, I'm thinking that `WithStatus` semantics may be a bit more desirable, thoughts?

Reply:
That makes sense, and I agree 💯